src/sentry_value.h

```diff
@@ -96,6 +97,18 @@ sentry_value_t sentry__value_clone(sentry_value_t value);
 int sentry__value_append_bounded(
     sentry_value_t value, sentry_value_t v, size_t max);
 
+/**
+ * Deep-merges two objects into each other.
```

Comment:
Nit: This does not actually merge the objects "into each other", it merges `src` into `dst`.